### PR TITLE
python api for PPMLContext

### PR DESCRIPTION
**Pull request by PatrickkZ**
1.init a PPMLConext(with or without parameters)
2.read csv file(plain text or encrypted file)
3.write to a csv file(plain text or encrypted file)

tested on

- python=3.6 & pyspark=2.4.6
- python=3.7 & pyspark=3.1.2

before run simple_query_example.py, make sure

1. run make-dist.sh at /BigDL/scala
2. run prepare_env.sh at /BigDL/python/ppml/dev
to set up the environment

then run `bigdl/ppml/api/simple_query_example.py`  with

python simple_query_example.py \
--simple_app_id 123456 \
--simple_app_key 123456\
--primary_key_path /your/primary/key/path/primaryKey \
--data_key_path /your/data/key/path/dataKey \
--input_path /your/test/data/path \
--output_path /your/output/path \
--input_encrypt_mode AES/CBC/PKCS5Padding or plain_text \
--output_encrypt_mode AES/CBC/PKCS5Padding or plain_text

### CODE REVIEW

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextWrapper.scala [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   */
+
+  def option(encryptedDataFrameReader: EncryptedDataFrameReader,
+             key: String, value: String): EncryptedDataFrameReader = {
+    logger.info("option...")
+    encryptedDataFrameReader.option(key, value)
+  }
+
+  def csv(encryptedDataFrameReader: EncryptedDataFrameReader, path: String): DataFrame = {
+    logger.info("csv...")
```
Comment (glorysdj) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextWrapper.scala:79]: debug log should be removed

Comment (glorysdj) [reply scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextWrapper.scala:79]: or keep them with logger.deug

File: python/ppml/src/bigdl/ppml/api/ppml_context.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+from bigdl.ppml.api import *
+import argparse
+
+
+class PPMLContext(JavaValue):
```
Comment (qiuxin2012) [inline python/ppml/src/bigdl/ppml/api/ppml_context.py:5]: move to bigdl/ppml

**Comment by qiuxin2012** [comment]
Add unit tests and enable them in jenkins

File: python/ppml/src/bigdl/ppml/api/__init__.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+from bigdl.dllib.utils.engine import prepare_env
+from bigdl.dllib.utils.common import *
+from bigdl.ppml.fl.utils import *
+from bigdl.ppml.ppml_context import *
+import bigdl
+
+import psutil
+
+total_mem = psutil.virtual_memory().total
+spark_mem_gb = int(total_mem / (2 ** 30) * 0.5)
```
Comment (qiuxin2012) [inline python/ppml/src/bigdl/ppml/api/__init__.py:28]: maybe we should read the value from env, use this default value when user doen't set SPARK_DRIVER_MEMORY

File: python/ppml/src/bigdl/ppml/ppml_context.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                             "Accepted save modes are 'overwrite', 'append', 'ignore', 'error', 'errorifexists'.")
+        self.df_writer = callBigDlFunc(self.bigdl_type, "mode", self.df_writer, mode)
+        return self
+
+    def csv(self, path):
+        return callBigDlFunc(self.bigdl_type, "csv", self.df_writer, path)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
```
Comment (qiuxin2012) [inline python/ppml/src/bigdl/ppml/ppml_context.py:74]: wrap this to a class.

File: python/ppml/test/bigdl/ppml/resources/people.csv [hunk 5]
```diff
@@ -0,0 +1,101 @@
+name,age,job
```
Comment (qiuxin2012) [inline python/ppml/test/bigdl/ppml/resources/people.csv:1]: You had better generate these files to a templete folder when you run your test, and delete them when the test finished.

File: python/ppml/src/bigdl/ppml/ppml_context.py (hunk 4 above)
Comment (PatrickkZ) [reply python/ppml/src/bigdl/ppml/ppml_context.py]: remove the usage example in `ppml_context.py` , and add a new file `simple_query_example.py`

File: python/ppml/src/bigdl/ppml/api/simple_query_example.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import argparse
+
+from bigdl.ppml.ppml_context import *
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--simple_app_id", type=str, required=True, help="simple app id")
+parser.add_argument("--simple_app_key", type=str, required=True, help="simple app key")
+parser.add_argument("--primary_key_path", type=str, required=True, help="primary key path")
+parser.add_argument("--data_key_path", type=str, required=True, help="data key path")
+parser.add_argument("--input_encrypt_mode", type=str, required=True, help="input encrypt mode")
```
Comment (qiuxin2012) [inline python/ppml/src/bigdl/ppml/api/simple_query_example.py:26]: do we need this?

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextWrapper.scala [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+import com.intel.analytics.bigdl.ppml.crypto.{AES_CBC_PKCS5PADDING, BigDLEncrypt, CryptoMode, ENCRYPT, EncryptRuntimeException}
+import com.intel.analytics.bigdl.ppml.crypto.dataframe.{EncryptedDataFrameReader, EncryptedDataFrameWriter}
+import com.intel.analytics.bigdl.ppml.kms.{KMS_CONVENTION, SimpleKeyManagementService}
+import org.apache.spark.sql.{DataFrame, DataFrameWriter, Row}
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.io.File
+import java.util
+
+object PPMLContextWrapper {
```
Comment (qiuxin2012) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextWrapper.scala:29]: rename to PPMLContextPython ?

**Review by glorysdj — APPROVED** [review]
LGTM

**Comment by PatrickkZ** [comment]
https://github.com/intel-analytics/BigDL/pull/4775

**Comment by qiuxin2012** [comment]
Python style check failed due to this PR.
https://github.com/intel-analytics/BigDL/pull/4880

**Pull request merged**